controller/main.go

```diff
@@ -163,8 +211,15 @@ func main() {
 		*namespace = string(bs)
 	}
 
+	poolStatusChan := make(chan event.GenericEvent)
+
 	c := &controller{
 		ips: allocator.New(),
+		generatePoolStatusEvent: func(name string) {
+			go func() {
```

Comment:
well, what I'm trying to do here is not to block in case there is any issue in the pool_status_controller (internally). If it is safe to accept that the handler of the controller runtime always reads this channel and queues the event, then it is safe to create this routine

Reply:
If spinning up the go routine is not necessary, we shouldn't have it. If the controller blocks, we have a problem anyway and having a go routine just mitigates it without really understanding (plus, we won't have guarantee on the order the events are generated).

Reply:
ok